lib/Console/Command/Parse.php

```diff
@@ -23,37 +23,93 @@
 
 use \Symfony\Component\Console\Input\InputInterface;
 use \Symfony\Component\Console\Output\OutputInterface;
+use \Mesamatrix\Parser\OglVersion;
+use \Mesamatrix\Parser\UrlCache;
+use \Mesamatrix\Parser\Hints;
 
 class Parse extends \Symfony\Component\Console\Command\Command
 {
-    protected function configure()
-    {
+    protected $urlCache;
+    protected $statuses;
+
+    protected function configure() {
         $this->setName('parse')
              ->setDescription('Parse data and generate XML')
         ;
     }
 
-    protected function execute(InputInterface $input, OutputInterface $output)
-    {
-        //$gitLog = exec_git("log --pretty=format:%H --reverse " .
-        //    Mesamatrix::$config->getValue("git", "oldest_commit").".. -- ".Mesamatrix::$config->getValue("git", "gl3"), $log);
-        //$initialCommit = rtrim(fgets($log));
-        $initialCommit = "master";
-
-        $cat = new \Mesamatrix\Git\ProcessBuilder(array(
-          'show', $initialCommit.':'.\Mesamatrix::$config->getValue('git', 'gl3')
+    protected function execute(InputInterface $input, OutputInterface $output) {
+        $gl3Path = \Mesamatrix::$config->getValue('git', 'gl3', 'docs/GL3.txt');
+        $gitLog = new \Mesamatrix\Git\ProcessBuilder(array(
+            'log', '--pretty=format:%H|%at|%aN|%cN', '--reverse',
+            \Mesamatrix::$config->getValue('git', 'oldest_commit').'..', '--',
+            $gl3Path
         ));
-        $proc = $cat->getProcess();
-        $this->getHelper('process')->mustRun($output, $proc);
-
-        $hints = new \Mesamatrix\Parser\Hints();
-        $parser = new \Mesamatrix\Parser\OglParser($hints);
-        $matrix = $parser->parse_content($proc->getOutput());
+        $gitLogProc = $gitLog->getProcess();
+        $this->getHelper('process')->mustRun($output, $gitLogProc);
+
+        $commitLines = explode(PHP_EOL, $gitLogProc->getOutput());
+        $hints = new Hints();
+        $matrix = new \Mesamatrix\Parser\OglMatrix();
+        $parser = new \Mesamatrix\Parser\OglParser($hints, $matrix);
+        foreach ($commitLines as $commitLine) {
+            list($commitHash, $time, $author, $committer) = explode('|', $commitLine);
+            $commit = new \Mesamatrix\Git\Commit($commitHash, $time, $author, $committer);
+            \Mesamatrix::$logger->info('Parsing GL3.txt for commit '.$commit->getHash());
+            $cat = new \Mesamatrix\Git\ProcessBuilder(array(
+              'show', $commit->getHash().':'.$gl3Path
+            ));
+            $proc = $cat->getProcess();
+            $this->getHelper('process')->mustRun($output, $proc);
+
+            $parser->parse_content($proc->getOutput(), $commit);
+        }
```

Comment:
And I still think `parse*` methods should not return the matrix, especially because it's not used anywhere. If it should return anything, it would be a boolean or a status code.